src/app/ui/input-keyvalue/input-keyvalue.component.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 Carnegie Mellon University. 
+// Released under a MIT (SEI) license. See LICENSE.md in the project root. 
+
+import { Component, OnInit, Input } from '@angular/core';
+import { AbstractControl, FormControl } from '@angular/forms';
+
+@Component({
+  selector: 'app-input-keyvalue',
+  templateUrl: './input-keyvalue.component.html',
+  styleUrls: ['./input-keyvalue.component.scss']
+})
+export class InputKeyValueComponent implements OnInit {
+  @Input() label = '';
+  @Input() field = '';
+  @Input() keyplaceholder = 'Key';
+  @Input() valueplaceholder = 'Value';
+  @Input() maxlength = 100;
+  @Input() control: AbstractControl;
+  keyfield = '';
+  valuefield = '';
+  items: Array<ListPair> = [];
+  keyinput = new FormControl('');
+  valueinput = new FormControl('');
+  
+  constructor() { 
+  }
+
+  ngOnInit(): void {
+    this.keyfield = this.field+'Key';
+    this.valuefield = this.field+'Value';
+  }
+
+  getPair(item: ListPair): string {
+    return `${item.type?.trim()} = ${item.value?.trim()}`;
+  }
+
+  addPair() {
+    this.control.value.push(
+      { 
+        type: this.keyinput.value,
+        value: this.valueinput.value
+      }
+    );
+    this.control.markAsDirty();
+    this.keyinput.reset();
+    this.valueinput.reset();
+  }
+
+  removePair(item: ListPair) {
+    item.deleted = !item.deleted;
+    this.control.markAsDirty();
+  }
+
+}
+
+export interface ListPair {
+  id?: string | number;
+  type?: string; // IdentityServer uses 'Type' as 'Key'
+  value?: string;
+  deleted?: boolean;
+}
```

Comment:
I tried to make a generic model for this component with `key` and `value` as attributes, but adding this does not seem to do much, and rather `control.value` seems to be an array of type `ClientClaim` at runtime with whatever attributes that model has, so I used `type` here to match ClientClaim.

Putting this ListPair model here doesn't seem to do anything. I used the same method used for the `input-list` component: https://github.com/cmu-sei/identity-ui/blob/65b93ef314cbb0dc0f08ace90a642783c98c2029/src/app/ui/input-list/input-list.component.ts#L40-L43
It seems to help add type checking to uses of the object, but in our case, it also forces us to use the interface of ClientClaim (with key being "type") which would mean any model that wants to use this component would need its key field to be "type". 

I'm assuming it's no big deal at the moment, but if we want to make this component more generic, I could allow one to specify the attribute name for "key" when using the component and then do `item['key_name']` when accessing the key...

Reply:
For example,
```
<app-input-keyvalue label="Claims" field="claims" keyname="type"
       maxlength="50" [control]="form.controls.claims">
</app-input-keyvalue>
```
```
export class InputKeyValueComponent implements OnInit {
      @Input() keyname = 'key';
      
      getPair(item: ListPair): string {
          return `${item[this.keyname]?.trim()} = ${item.value?.trim()}`;
      }
}
export interface ListPair {
    id?: string | number;
    value?: string;
    deleted?: boolean;
}
```
This seems to work pretty well and is more general for any key/value pair we might want (if some don't have type as the key name). But it's a little bit more complicated to use.